cmd/kaspawallet/daemon/server/sync.go

```diff
@@ -0,0 +1,257 @@
+package server
+
+import (
+	"github.com/kaspanet/kaspad/app/appmessage"
+	"github.com/kaspanet/kaspad/domain/consensus/model/externalapi"
+	"github.com/pkg/errors"
+	"time"
+)
+
+const (
+	externalKeychain = 0
+	internalKeychain = 1
+)
+
+var keyChains = []uint8{externalKeychain, internalKeychain}
+
+func (s *server) sync() error {
+	ticker := time.NewTicker(time.Second)
+	defer ticker.Stop()
+
+	for range ticker.C {
+		err := s.collectCloseUTXOs()
+		if err != nil {
+			return err
+		}
+
+		err = s.collectFarUTXOs()
+		if err != nil {
+			return err
+		}
+
+		err = s.refreshExistingUTXOsWithLock()
```

Comment:
In collectUTXOsFromRecentAddresses I release the lock between calls to the RPC server. I think it's nice because it gives the user more opportunities to have its request in the head of the queue

Reply:
How much time does collectUTXOsFromRecentAddresses take?
If less then a second - I'd argue that simpler sync-code is more important, if not - I'd agree with you.

Reply:
It depends on the last used index. But for indexes above 10k it can take a few seconds